### PR TITLE
Fire - Add ability to disable screams for individual units

### PR DESCRIPTION
**Pull request by DartRuffian**
**When merged this pull request will:**
- _Check for the_ `ace_fire_enableScreams` _variable as well as the setting before playing a scream sound._

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: docs/wiki/framework/fire-framework.md [hunk 1]
```diff
@@ -37,3 +37,12 @@ Use `CBA_fnc_serverEvent` to use the following features. Events are defined only
    | Arguments | Type | Optional (default value)
 ---| --------- | ---- | ------------------------
 0  | Fire source ID | Any  | Required
+
+
+## 2. Variables
+
+Screams can be disabled for an individual unit by setting the `ace_fire_enableScreams` variable on the unit, which should be synced across machines.
```
[the remark this answers is not in the document]
Comment (DartRuffian) [reply docs/wiki/framework/fire-framework.md:44]: I guess I missed the `// isGlobal` comment and looked at the default value.

Had a bit of a weird use-case, wanted to disable the ace_fire screams for specific units while using another mod.

[the remark this answers is not in the document]
Comment (johnb432) [reply docs/wiki/framework/fire-framework.md:44]: > Had a bit of a weird use-case, wanted to disable the ace_fire screams for specific units while using another mod.

Sounds fair enough.

[the remark this answers is not in the document]
Comment (johnb432) [reply docs/wiki/framework/fire-framework.md:44]: Imo `should` is too strong.

```suggestion
Screams can be disabled for an individual unit by setting the `ace_fire_enableScreams` variable on the unit, which can be synced across machines.
```

File: docs/wiki/framework/fire-framework.md [hunk 2]
```diff
@@ -37,3 +37,12 @@ Use `CBA_fnc_serverEvent` to use the following features. Events are defined only
...
+
+```sqf
+_unit setVariable ["ace_fire_enableScreams", false, true];
```
Comment (johnb432) [inline docs/wiki/framework/fire-framework.md:47]:
```suggestion
_unit setVariable ["ace_fire_enableScreams", false, _isGlobal];
```

File: addons/fire/XEH_postInit.sqf [hunk 3]
```diff
@@ -3,8 +3,8 @@
 [QGVAR(burn), FUNC(burn)] call CBA_fnc_addEventHandler;
 [QGVAR(playScream), {
     params ["_scream", "_source"];
-    // only play sound if enabled in settings
-    if (GVAR(enableScreams)) then {
+    // only play sound if enabled in settings and enabled for the unit
+    if (GVAR(enableScreams) and {_source getVariable [QGVAR(enableScreams), true]}) then {
```
Comment (johnb432) [inline addons/fire/XEH_postInit.sqf:7]:
```suggestion
    if (GVAR(enableScreams) && {_source getVariable [QGVAR(enableScreams), true]}) then {
```

**Review by johnb432 — APPROVED** [review]

**Pull request merged**